src/commands/community/nft/query.ts

```diff
@@ -130,16 +136,45 @@ export async function composeNFTDetail(
       })
     : []
 
-  const embed = composeEmbedMessage(msg, {
+  let embed = composeEmbedMessage(msg, {
     author: [
       `${capitalizeFirst(colName)}${chainName ? ` (${chainName})` : ""}`,
       ...(colImage.length ? [colImage] : []),
     ],
     description,
     image: nftImage,
     color: rarityColors[rarity?.rarity?.toUpperCase()],
-  }).addFields(fields)
-  return justifyEmbedFields(embed, 3)
+  }).addFields(attributeFields)
+
+  embed = justifyEmbedFields(embed, 3)
+
+  const activityRes = await community.getNFTActivity({
+    collectionAddress: collection_address,
+    tokenId: token_id,
+  })
+  if (activityRes.ok) {
+    const txHistoryTitle = `${getEmoji("swap")} Transaction History`
+    const txHistoryValue = (activityRes.data.data ?? [])
+      .map((tx) => {
+        if (!tx) return
```

Comment:
use `filter()` to exclude `undefined` items created here

Reply:
tui bỏ khúc check undefined rồi nha. Mới check lại không bị undefined